### DISCUSSION TITLE
Intent Issues

### DISCUSSION THREAD

**Discussion by ptyork** [OpenVoiceOS/ovos-buildroot#177] [Q&A] [answered]
I've been having a great time getting to know this project. Thanks so much for the amazing work.

I'm having some issues with and questions about intent parsing for a custom skill. Just placing it in the `.local/share/mycroft/skills` directory for now.

 1. With intent files, I can't get parenthetical expansion to work. E.g., 'hello there' and 'hi there' would both work separately, but '(hello | hi) there' would not. I feel like there must be some kind of preprocessing step that I haven't enabled. Where should I look for a solution?
 2. I'm just wanting to use the "headless stack" (bus, skills, audio, voice) to serve as the voice assistant for a custom system. So things like a file browser & media player are not desirable. Especially as the media player seems to hijack any unknown utterance, spin for 20+ seconds, and then saying "Sorry, I don't know how to play [last_word_in_utterance]". I uninstalled filebrowser, commonplay...any hidden adverse consequences? Especially with filebrowser as it was in `requirements/skills-essential.txt` for ovos-core.
 3. I'm not understanding fallback skills. If an utterance is not recognized, the intent service appears to try to collect fallback skills twice. Then it says skill-ovos-fallback-unknown will try to handle it (but it seems not to, unfortunately), and then it again checks twice to find fallback candidates. I just want an error tone or brief message played if the utterance is not recognized. What am I doing wrong?
 4. The system seems to want to refresh the nltk_data package every time it calls handle_utterance. Is this needed or a (mostly harmless) bug? I will need for this to work entirely offline often, so I'm hoping this isn't a requirement (at least after an initial download / setup).

Thanks again!!

**Comment by JarbasAl** [comment, marked as answer]
> I've been having a great time getting to know this project. Thanks so much for the amazing work.
> 
> I'm having some issues with and questions about intent parsing for a custom skill. Just placing it in the `.local/share/mycroft/skills` directory for now.
> 
>     1. With intent files, I can't get parenthetical expansion to work. E.g., 'hello there' and 'hi there' would both work separately, but '(hello | hi) there' would not. I feel like there must be some kind of preprocessing step that I haven't enabled. Where should I look for a solution?
> 
this should work out of the box and there is no config option to disable it, if that is not working you seem to have found a bug

>     2. I'm just wanting to use the "headless stack" (bus, skills, audio, voice) to serve as the voice assistant for a custom system. So things like a file browser & media player are not desirable. Especially as the media player seems to hijack any unknown utterance, spin for 20+ seconds, and then saying "Sorry, I don't know how to play [last_word_in_utterance]". I uninstalled filebrowser, commonplay...any hidden adverse consequences? Especially with filebrowser as it was in `requirements/skills-essential.txt` for ovos-core.
> 

no skill is mandatory, the list of skills is just a recommended list that will reflect what most users expect a VA to do out of the box

OCP is not a skill, it provides intent but it's part of the audio-stack, it can not be disabled

hijacking any unknown utterance is an issue with the padatious intent parser itself, the good news is that you can disable it, padatious is no longer used by default since https://github.com/OpenVoiceOS/ovos-config/pull/50

>     3. I'm not understanding fallback skills. If an utterance is not recognized, the intent service appears to try to collect fallback skills twice. Then it says skill-ovos-fallback-unknown will try to handle it (but it seems not to, unfortunately), and then it again checks twice to find fallback candidates. I just want an error tone or brief message played if the utterance is not recognized. What am I doing wrong?
> 
the logs need some improvements, it appears to check twice but it is doing so at different confidence levels, the error tone happens if no fallback handled the utterance, this is weird if you have fallback-unknown installed (that skill should always answer)

>     4. The system seems to want to refresh the nltk_data package every time it calls handle_utterance. Is this needed or a (mostly harmless) bug? I will need for this to work entirely offline often, so I'm hoping this isn't a requirement (at least after an initial download / setup).
> 
> 
it's a harmless bug, nltk is used by some plugins that need to download things and the logs are spammy
